src/physics/cam/cam_diagnostics.F90

```diff
@@ -1415,6 +1422,65 @@ subroutine diag_phys_writeout_moist(state, pbuf, p_surf_t)
     end do
     call outfld ('ATMEINT   ', ftem(:ncol,1), ncol, lchnk)
 
+    ! Add energy transport diagnostics
+    !
+    do k = 1, pver
+      z3(:ncol,k) = state%zm(:ncol,k) + state%phis(:ncol)*rga
+    end do
+
+    !! calculate the atmospheric energy transport terms:
+    !! internal energy transport Cp*T*v*dp/g
+    ftem(:ncol,:) = (cpair*state%t(:ncol,:)*state%v(:ncol,:))*(state%pdel(:ncol,:)/gravit)
+    !call outfld ('INTEnTrans', ftem(:ncol,:)  ,ncol   ,lchnk     )
+    !! vertically integrate
+    ftem3(:ncol)=ftem(:ncol,1)
+    do k=2,pver
+      ftem3(:ncol) = ftem3(:ncol) + ftem(:ncol,k)
+    end do
+    call outfld ('IntHeatTr', ftem3(:ncol), ncol   ,lchnk)
```

Comment:
The `ftem` calculation is fairly expensive and should only be done if `IntHeatTr` will be output. Please implement that as shown in this suggestion (and also below with the appropriate history field names).
```suggestion
    if (hist_field_active('INTEnTrans')) then
      ftem(:ncol,:) = (cpair*state%t(:ncol,:)*state%v(:ncol,:))*(state%pdel(:ncol,:)/gravit)
    !call outfld ('INTEnTrans', ftem(:ncol,:)  ,ncol   ,lchnk     )
      !! vertically integrate
      ftem3(:ncol)=ftem(:ncol,1)
      do k=2,pver
        ftem3(:ncol) = ftem3(:ncol) + ftem(:ncol,k)
      end do
      call outfld ('IntHeatTr', ftem3(:ncol), ncol   ,lchnk)
    end if
```